app/models/datacite_endpoint.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class DataCiteEndpoint < ::Endpoint
+  store :options, accessors: %i[mode prefix username password]
+
+  def switch!
+    Hyrax::DOI::DataCiteRegistrar.mode = mode
+    Hyrax::DOI::DataCiteRegistrar.prefix = prefix
+    Hyrax::DOI::DataCiteRegistrar.username = username
+    Hyrax::DOI::DataCiteRegistrar.password = password
+  end
+
+  # Remove the solr collection then destroy this record
+  def remove!
+    destroy
```

Comment:
What destroy is this calling? This method seems to be missing a test.

Reply:
@BertZZ  it is calling activerecord destroy.
https://apidock.com/rails/ActiveRecord/Relation/destroy

Reply:
I was lazy and wasn't really writing tests beyond what hyku was already testing.  I'm not entirely sure if this method is being used but left it in the copy from solr endpoint.  (Looks like I need to fix that comment at the least.)  I'll do a quick look and see if I can remove it or add a test if it is still necessary.

Reply:
@mankind That method takes an id usually which is what thew me here.

Reply:
@BertZZ not always. You can fetch the object and just call destroy on it with no arguments.

Reply:
I don't think the method is actually called for DataCiteEndpoint but I left it in to match the other endpoints.  Tests added including testing that the destroy cascades from the account destroy since I didn't override CleanupAccountJob to call `remove!` explicitly for DataCiteEndpoint.